nemo/collections/asr/models/aed_multitask_models.py

```diff
@@ -95,83 +95,82 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
-                mask_future=False,
-                num_attention_heads=transf_encoder_cfg_dict['num_attention_heads'],
-                attn_score_dropout=transf_encoder_cfg_dict['attn_score_dropout'],
-                attn_layer_dropout=transf_encoder_cfg_dict['attn_layer_dropout'],
-                ffn_dropout=transf_encoder_cfg_dict['ffn_dropout'],
-                pre_ln=transf_encoder_cfg_dict.get('pre_ln', True),
-                pre_ln_final_layer_norm=transf_encoder_cfg_dict.get('pre_ln_final_layer_norm', True),
-            )
+            self.transf_encoder = EncDecMultiTaskModel.from_config_dict(transf_encoder_cfg_dict)
             std_init_range = 1 / transf_encoder_cfg_dict['hidden_size'] ** 0.5
             self.transf_encoder.apply(lambda module: transformer_weights_init(module, std_init_range))
 
-        transf_decoder_cfg_dict = OmegaConf.to_container(cfg.get('transf_decoder'))
+        transf_decoder_cfg_dict = cfg.transf_decoder
 
         # Transformer decoder
         vocab_size = 8 * ceil(self.tokenizer.vocab_size / 8)
-        transf_decoder_cfg_dict['vocab_size'] = vocab_size
-        library = transf_decoder_cfg_dict.pop('library', 'nemo')
-        model_name = transf_decoder_cfg_dict.pop('model_name', None)
-        pretrained = transf_decoder_cfg_dict.pop('pretrained', False)
-        self.transf_decoder = get_transformer(
-            library=library,
-            model_name=model_name,
-            pretrained=pretrained,
-            config_dict=transf_decoder_cfg_dict,
-            encoder=False,
-            pre_ln_final_layer_norm=transf_decoder_cfg_dict.get("pre_ln_final_layer_norm", False),
-        )
 
-        self.log_softmax = TokenClassifier(
-            hidden_size=self.transf_decoder.hidden_size,
-            num_classes=vocab_size,
-            activation=self.cfg.head.activation,
-            log_softmax=self.cfg.head.log_softmax,
-            dropout=self.cfg.head.dropout,
-            use_transformer_init=self.cfg.head.use_transformer_init,
-        )
-        self.log_softmax.mlp.layer0.weight = self.transf_decoder.embedding.token_embedding.weight
+        # Auto inject vocab size for `get_transformer`
+        with open_dict(transf_decoder_cfg_dict):
+            if 'config_dict' in transf_decoder_cfg_dict:
+                transf_decoder_cfg_dict['config_dict']['vocab_size'] = vocab_size
+
+        self.transf_decoder = EncDecMultiTaskModel.from_config_dict(transf_decoder_cfg_dict)
+
+        # Setup token classifier
+        with open_dict(self.cfg.head):
+            self.cfg.head.num_classes = vocab_size
+
+        self.log_softmax = EncDecMultiTaskModel.from_config_dict(self.cfg.head)
+        if isinstance(self.log_softmax, MultiLayerPerceptron):
+            self.log_softmax.layer0.weight = self.transf_decoder.embedding.token_embedding.weight
+        else:
+            raise ValueError(f"Expected MultiLayerPerceptron head, got {type(self.log_softmax)}")
```

Comment:
Shall we force the classifier to `MultiLayerPerceptron`? Putting it in config may caused errors by the users. @titu1994